src/impy/models/pythia8.py

```diff
@@ -56,46 +61,38 @@ def __init__(self, event_kinematics, seed=None, logfname=None):
         self._lib.pythia = self._lib.Pythia(datdir, True)
 
         # must come last
-        self.event_kinematics = event_kinematics
+        self.kinematics = evt_kin
         self._set_final_state_particles()
 
-    def _cross_section(self, evt_kin):
-        with self._temporary_evt_kin(evt_kin):
-            st = self._lib.pythia.info.sigmaTot
-            return CrossSectionData(
-                st.sigmaTot,
-                st.sigmaTot - st.sigmaEl,
-                st.sigmaEl,
-                st.sigmaAX,
-                st.sigmaXB,
-                st.sigmaXX,
-                st.sigmaAXB,
-            )
+    def _cross_section(self):
+        st = self._lib.pythia.info.sigmaTot
+        return CrossSectionData(
+            st.sigmaTot,
+            st.sigmaTot - st.sigmaEl,
+            st.sigmaEl,
+            st.sigmaAX,
+            st.sigmaXB,
+            st.sigmaXX,
+            st.sigmaAXB,
+        )
 
-    def _set_event_kinematics(self, k):
+    def _set_kinematics(self, kin):
         pythia = self._lib.pythia
         pythia.settings.resetAll()
 
-        if not pythia.particleData.isParticle(k.p1pdg):
-            raise ValueError(f"Particle {k.p1pdg} not recognized")
-
-        if not pythia.particleData.isParticle(k.p2pdg):
-            raise ValueError(f"Particle {k.p2pdg} not recognized")
-
         config = [
             "Random:setSeed = on",
             f"Random:seed = {self._seed}",
-            # Specify energy in center of mass
+            # use center-of-mass frame
             "Beams:frameType = 1",
             "SoftQCD:inelastic = on",
+            "PhotonParton:all = on",
```

Comment:
Which process does require this setting to be true?

Reply:
Currently nothing, but I would like to enable photo-hadron interactions for Pythia8.

Reply:
I don't recommend setting this to on because it [restricts the available PDF](https://pythia.org/latest-manual/ElectroweakProcesses.html#section2) selection and requires some thinking by the user what they want to do. The default real photon collisions should work without this flag since the relevant switch `Photon:ProcessType` is on by default.

Reply:
Seems you are right. I thought I tested photons without this and got failures, but I just checked that it works without. I will remove this line then.